### PR TITLE
Posts list powered by DataViews

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Part of: https://github.com/WordPress/gutenberg/issues/62371

This PR explores the addition of Data Views for the new experimental `posts` list under `Gutenberg->posts` menu.
This is just a starting point, but wanted to have it on GH to start some discussions.

What this PR does for now is the start of making an abstraction that could be used for any post type, but we need to do it at least for `posts` and `pages` for now.

Maybe the notable things for now to discuss are:
1. When we'll choose to edit an item, should it load the `post` or the `site` editor? I've started by having as a goal to load the site editor. This has nuances though. Do we want to have for example global styles in a the editor of the post? Additionally the editor adds some extra settings during initialization (`initializeEditor`) that are needed here too. How can we abstract this to be reused what's needed.
2. Design wise, we're also aiming towards having the `categories, tags`, etc.. in the sidebar where we would load the lists for the taxonomies. How does this affect the current `views` we have(`drafts, published, etc..`)? Should we consider changing placement of these views into the frame, which might also simplify some shared information that were wanted for views. An example is the number of items per view, which is not possible right now.

### Notes
1. We would need a similar API with `usePostActions` for fields. Now we are reusing the exact same code for the `posts` list and the existing `pages` list in site editor. So testing the `pages` list that works as before is also part of this PR to avoid regressions.

### CODE REVIEW

File: packages/edit-site/src/components/posts-app/index.js [hunk 1]
```diff
@@ -10,29 +10,27 @@ import { privateApis as routerPrivateApis } from '@wordpress/router';
...
-	areas: {
-		sidebar: 'Empty Sidebar',
-		content: <Page>Welcome to Posts</Page>,
-		preview: undefined,
-		mobile: <Page>Welcome to Posts</Page>,
-	},
-};
+function PostsLayout() {
+	// This ensures the edited entity id and type are initialized properly.
+	useInitEditedEntityFromURL();
```
Comment (youknowriad) [inline packages/edit-site/src/components/posts-app/index.js:23]: How hard would it be to move away from this hook and instead of having postId postType and context in the site editor store, just pass it down as props to the `Editor` component. (For the site editor, not the posts page, we could keep syncing these to the store for backwards compatibility)

File: packages/edit-site/src/components/page-pages/style.scss [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// TODO: we need to examine if these styles could be the default styles
+// for every post type and/or how to have different styles for different post types.
+// Example is `posts` and `pages`.
```
Comment (youknowriad) [inline packages/edit-site/src/components/page-pages/style.scss]: The classnames seems wrong now right?

File: packages/edit-site/src/components/layout/router.js [hunk 3]
```diff
@@ -92,15 +92,15 @@ export default function useLayoutAreas() {
...
+				content: <PostsList postType={ postType } />,
 				preview: ( isListLayout || canvas === 'edit' ) && (
 					<Editor isLoading={ isSiteEditorLoading } />
 				),
 				mobile:
 					canvas === 'edit' ? (
 						<Editor isLoading={ isSiteEditorLoading } />
 					) : (
-						<PagePages />
+						<PostsList postType={ postType } />
```
Comment (youknowriad) [inline packages/edit-site/src/components/layout/router.js:99]: This is the kind of change I was hoping to see :)

File: packages/edit-site/src/components/posts-app/router.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	const { postType, layout, canvas } = params;
+	const labels = useSelect(
+		( select ) => {
+			return select( coreStore ).getPostType( postType )?.labels;
+		},
+		[ postType ]
+	);
+
+	// Posts list.
+	if ( [ 'page', 'post' ].includes( postType ) ) {
```
Comment (youknowriad) [inline packages/edit-site/src/components/posts-app/router.js]: Why page?

File: packages/edit-site/src/components/posts-app/router.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import DataViewsSidebarContent from '../sidebar-dataviews';
+import PostsList from '../posts-app/posts-list';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
+	const { params = {} } = useLocation();
+	const { postType, layout, canvas } = params;
+	const labels = useSelect(
```
Comment (youknowriad) [inline packages/edit-site/src/components/posts-app/router.js:23]: I would prefer if we avoid useSelect and complex logic in the "router". Ideally this is actually just a global object (when we'll be able to define "paths" as part of the config). So I'd actually move these calls to dedicated components that I render in areas.

File: packages/edit-site/src/components/posts-app/router.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+import Editor from '../editor';
+import SidebarNavigationScreen from '../sidebar-navigation-screen';
+import SidebarNavigationScreenMain from '../sidebar-navigation-screen-main';
+import DataViewsSidebarContent from '../sidebar-dataviews';
+import PostsList from '../posts-app/posts-list';
+
+const { useLocation } = unlock( routerPrivateApis );
+
+export default function useLayoutAreas() {
+	const isSiteEditorLoading = useIsSiteEditorLoading();
```
Comment (youknowriad) [inline packages/edit-site/src/components/posts-app/router.js]: Already existing in the other router, but maybe we can just move this to the "Editor" component now?

**Review by youknowriad — COMMENTED** [review]
Great start

File: packages/edit-site/src/components/posts-app/router.js (hunk 4 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/posts-app/router.js]: Just a lot of temp code to get first feedback. These should be generic eventually and we could even restrict this for `posts` as a first step..

File: packages/edit-site/src/components/posts-app/router.js (hunk 6 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/posts-app/router.js]: Yeah, I have this in mind for things that end up being reused, to be moved elsewhere.. Just temp code for first feedback.

**Comment by youknowriad** [comment]
> When we'll choose to edit an item, should it load the post or the site editor? I've started by having as a goal to load the site editor. This has nuances though. Do we want to have for example global styles in a the editor of the post? Additionally the editor adds some extra settings during initialization (initializeEditor) that are needed here too. How can we abstract this to be reused what's needed.

Good question, I'm thinking we use the same "component", it doesn't matter if we call it post or site editor. That component will probably be the site editor one for a start because we're building this stuff in the site editor package today. The Global Styles question is a good one, I don't think we need to get everything right in the first PR. I think it's fine to keep it as is for now.

> Design wise, we're also aiming towards having the categories, tags, etc.. in the sidebar where we would load the lists for the taxonomies. How does this affect the current views we have(drafts, published, etc..)? Should we consider changing placement of these views into the frame, which might also simplify some shared information that were wanted for views. An example is the number of items per view, which is not possible right now.

Good question as well, maybe "posts" is a top level menu item and you gave to go into it first (like pages in the site editor) but it raises the question of what is the "front page" of this experience. Also a design question mainly that can be addressed in a follow-up.

> We would need a similar API with usePostActions for fields. Now we are reusing the exact same code for the posts list and the existing pages list in site editor. So testing the pages list that works as before is also part of this PR to avoid regressions.

Definitely 💯

**Comment by jasmussen** [comment]
> Good question as well, maybe "posts" is a top level menu item and you gave to go into it first (like pages in the site editor) but it raises the question of what is the "front page" of this experience. Also a design question mainly that can be addressed in a follow-up.

That's the key bit. It can live here fo now:

Screenshot 2024-06-21 at 09 56 13

But in order to solve the structural questions it does need to move to the top level.

It's a good start, but I am seeing some errors, for awareness:
2024-06-21 09 54 55 local-wordpress local f52f2822480a

File: packages/edit-site/src/components/posts-app/posts-list.js [hunk 7]
```diff
@@ -293,13 +292,18 @@ export default function PagePages() {
...
 	const {
-		records: pages,
+		records,
 		isResolving: isLoadingPages,
 		totalItems,
 		totalPages,
 	} = useEntityRecords( 'postType', postType, queryArgs );
 
-	usePostIdLinkInSelection( selection, setSelection, isLoadingPages, pages );
+	usePostIdLinkInSelection(
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/posts-app/posts-list.js]: This can now be removed.

File: packages/edit-site/src/components/posts-app/posts-list.js [hunk 8]
```diff
@@ -312,20 +316,26 @@ export default function PagePages() {
...
 			const { getEntityRecord, getPostType, canUser } =
 				select( coreStore );
 			const siteSettings = getEntityRecord( 'root', 'site' );
 			return {
 				frontPageId: siteSettings?.page_on_front,
 				postsPageId: siteSettings?.page_for_posts,
-				addNewLabel: getPostType( 'page' )?.labels?.add_new_item,
-				canCreatePage: canUser( 'create', 'pages' ),
+				labels: getPostType( postType )?.labels,
+				// TODO: check what is the proper way to make this work for any post type..
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/posts-app/posts-list.js]: To get the resource (posts, pages etc) we can do this:
```
const postTypeObject = getPostType( postType );
const resource = postTypeObject?.rest_base || '';
```

File: packages/edit-site/src/components/sidebar-dataviews/default-views.js [hunk 9]
```diff
@@ -54,103 +58,114 @@ const DEFAULT_PAGE_BASE = {
...
-			view: {
-				...DEFAULT_PAGE_BASE,
-				filters: [
-					{
-						field: 'status',
-						operator: OPERATOR_IS_ANY,
-						value: 'future',
+				},
+				{
+					title: __( 'Scheduled' ),
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/sidebar-dataviews/default-views.js:94]: We are repeating the status view definition all the time I guess we can do something like:

```
const createStatusView = (title, slug, icon, status) => ({
  title,
  slug,
  icon,
  view: {
    ...DEFAULT_POST_BASE,
    filters: [
      {
        field: 'status',
        operator: OPERATOR_IS_ANY,
        value: status,
      },
    ],
  },
});

// Then use it like:
createStatusView(__('Published'), 'published', published, 'publish'),
createStatusView(__('Drafts'), 'drafts', drafts, 'draft'),
// ... and so on
```

**Comment by annezazu** [comment]
I was trying to test this today and got an error:

Screenshot 2024-06-26 at 1 36 45 PM

>TypeError: k.styles is not iterable
    at https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/edit-site/index.min.js?ver=29fd8339093b1150d2ff:72:3791
    at Object.useMemo (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:66553)
    at t.useMemo (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react.min.js?ver=18:2:6087)
    at cg (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/edit-site/index.min.js?ver=29fd8339093b1150d2ff:72:3779)
    at hu (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:60255)
    at xi (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:119196)
    at bs (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:108448)
    at vs (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:108376)
    at gs (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:108239)
    at as (https://playground.wordpress.net/scope:0.5644724289443985/wp-content/plugins/gutenberg/build/vendors/react-dom.min.js?ver=18:2:105074)

Minor too but this should be "Add post" rather than "Add page":
Screenshot 2024-06-26 at 1 36 52 PM

**Comment by ntsekouras** [comment]
>I was trying to test this today and got an error:

@annezazu it's being tracked and in progress. I only hid your comment because I'm expecting too many of them in this PR and had big screenshots 😄 .

File: packages/edit-site/src/posts.js [hunk 10]
```diff
@@ -11,16 +30,67 @@ import PostsApp from './components/posts-app';
...
+	// This is needed because edit-site doesn't initialize with EditorProvider,
+	// which internally uses updateEditorSettings as well.
+	dispatch( editorStore ).updateEditorSettings( {
+		defaultTemplateTypes: settings.defaultTemplateTypes,
+		defaultTemplatePartAreas: settings.defaultTemplatePartAreas,
+	} );
+
+	// Prevent the default browser action for files dropped outside of dropzones.
+	window.addEventListener( 'dragover', ( e ) => e.preventDefault(), false );
+	window.addEventListener( 'drop', ( e ) => e.preventDefault(), false );
```
Comment (youknowriad) [inline packages/edit-site/src/posts.js:81]: Feels like too much boilerplate. It's ok for now but might be good to take a look at this at some point to see where these things should be moved (or removed)

File: packages/edit-site/src/posts.js [hunk 11]
```diff
@@ -11,16 +30,67 @@ import PostsApp from './components/posts-app';
 
 /**
  * Initializes the "Posts Dashboard"
- * @param {string} id DOM element id.
+ * @param {string} id       ID of the root element to render the screen in.
+ * @param {Object} settings Editor settings.
  */
-export function initializePostsDashboard( id ) {
+export function initializePostsDashboard( id, settings ) {
```
Comment (youknowriad) [inline packages/edit-site/src/posts.js:34]: Can we avoid using the settings from the server? I would love if we can fetch all the necessary settings using REST API instead.

File: packages/edit-site/src/posts.js (hunk 10 above)
Comment (ntsekouras) [reply packages/edit-site/src/posts.js:81]: Agreed. I'm just looking at site editor load(PHP + JS) and will need to investigate what are the things we need.

File: packages/edit-site/src/posts.js (hunk 11 above)
Comment (ntsekouras) [reply packages/edit-site/src/posts.js:34]: I'll explore this, yes. I'm wondering though why we have it like this in site editor for now.. Maybe the endpoint is missing some settings 🤔 ?

Comment (youknowriad) [reply packages/edit-site/src/posts.js:34]: There's a lot of things that get through that settings object both in post and site editors and yeah initially the endpoints didn't have most of the settings. I'm fine with you did here for now. But let's track this. I think we should look at each setting separately and see how to avoid it.

File: packages/edit-site/src/components/sidebar-dataviews/default-views.js (hunk 9 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/sidebar-dataviews/default-views.js:94]: I don't think we gain much for now to do this change. Maybe see how to best refactor this code when we work more on `views`.

File: packages/edit-site/src/components/posts-app/posts-list.js (hunk 8 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/posts-app/posts-list.js]: Thanks Jorge!

**Comment by ntsekouras** [comment]
@jorgefilipecosta can you help me figure out why some actions like the `trashPostAction` don't appear for `posts` list? I need to check more the actions and it seems there are some differences between `post and page`, but I wasn't able to figure it out yet.

Additionally I observed that when we don't have any bulk actions and therefore no `checkbox`, at least the `table` view renders weird, because we have attached styles to that checkbox. We should probably finds a more resilient way to have good default styles. --cc @jameskoster 

Finally a last thing that maybe Jay can help is that I couldn't find why in `posts` list and `grid` view, the title is not aligned vertically..

**Comment by jorgefilipecosta** [comment]
Sure @ntsekouras, I'm having a look into the actions issue.

**Comment by jameskoster** [comment]
> Additionally I observed that when we don't have any bulk actions and therefore no checkbox, at least the table view renders weird, because we have attached styles to that checkbox. We should probably finds a more resilient way to have good default styles

I pushed a fix for that one.

**Comment by jameskoster** [comment]
> Finally a last thing that maybe Jay can help is that I couldn't find why in posts list and grid view, the title is not aligned vertically..

It's because `dataviews-view-grid__primary-field` has a different `display` property in the posts list. It should be `flex` but here it is `block`. 

Edit: Pushed a fix for this one too.

File: lib/experimental/posts/load.php [hunk 12]
```diff
@@ -7,22 +7,95 @@
 
 add_action( 'admin_menu', 'gutenberg_replace_posts_dashboard' );
 
+// Default to is-fullscreen-mode to avoid jumps in the UI.
+add_filter(
+	'admin_body_class',
+	static function ( $classes ) {
```
Comment (jorgefilipecosta) [inline lib/experimental/posts/load.php:13]: Normally anonymous functions are not recommended in core, because they don't allow a third-party plugin to use remove_filter.

File: lib/experimental/posts/load.php [hunk 13]
```diff
@@ -7,22 +7,95 @@
...
+		return "$classes is-fullscreen-mode";
+	}
+);
+
 /**
  * Renders the new posts dashboard page.
  */
 function gutenberg_posts_dashboard() {
+	// Flag that we're loading the block editor.
+	// $current_screen = get_current_screen();
```
Comment (jorgefilipecosta) [inline lib/experimental/posts/load.php]: Are these code comments still required?

Comment (ntsekouras) [reply lib/experimental/posts/load.php]: Still need to do at least a first good pass of needed settings. I'll update soon.

File: lib/experimental/posts/load.php (hunk 12 above)
Comment (ntsekouras) [reply lib/experimental/posts/load.php:13]: That's what we do for site editor in core right now. I'll have to check if I can add the filter only in this new page though.. I didn't observe something weird so far though..

File: lib/experimental/posts/load.php [hunk 14]
```diff
@@ -7,22 +7,95 @@
...
+	);
 	wp_enqueue_script( 'wp-edit-site' );
 
+	wp_enqueue_media();
+
 	echo '<div id="gutenberg-posts-dashboard"></div>';
 }
 
+/**
+ * Redirects to the new posts dashboard page and adds the postType query arg.
```
Comment (jorgefilipecosta) [inline lib/experimental/posts/load.php:71]: Instead of doing a redirect to include the post type. Could we assume a default post-type? e.g: if empty( $_GET['postType'] ) we assume it has the value post on our code? 
The current post list seems to work like that it does not have a postType parameter in case of post.

File: packages/dataviews/src/dataviews.tsx [hunk 15]
```diff
@@ -51,22 +54,6 @@ const defaultGetItemId = ( item: ItemWithId ) => item.id;
 
 const defaultOnSelectionChange = () => {};
 
-function useSomeItemHasAPossibleBulkAction< Item >(
```
Comment (jorgefilipecosta) [inline packages/dataviews/src/dataviews.tsx:54]: The change does not seem related. Probably it was a rebase issue.

**Review by jorgefilipecosta — COMMENTED** [review]
The back button is not working:
Screenshot 2024-06-28 at 10 55 11

File: lib/experimental/posts/load.php (hunk 12 above)
Comment (jorgefilipecosta) [reply lib/experimental/posts/load.php:13]: Maybe the current one for site editor, passed during a code review but as far as I know anonymous functions should be avoided, they work well the issue is if a plugin wants to remove the filter the plugin can not do it.

File: packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js [hunk 16]
```diff
@@ -213,7 +213,8 @@ function useResolveEditedEntityAndContext( { postId, postType } ) {
 		if ( postType && postId && authorizedPostTypes.includes( postType ) ) {
 			return { postType, postId };
 		}
-
+		// TODO: for post types lists we should probably not render the front page, but maybe a placeholder
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/components/sync-state-with-url/use-init-edited-entity-from-url.js:216]: I guess for other post types, we should show the list page for that post type instead of the front page. For example, for posts we could show the blog front page.

File: packages/edit-site/src/posts.js [hunk 17]
```diff
@@ -1,24 +1,96 @@
...
 	if ( ! globalThis.IS_GUTENBERG_PLUGIN ) {
 		return;
 	}
 	const target = document.getElementById( id );
 	const root = createRoot( target );
 
+	dispatch( blocksStore ).reapplyBlockTypeFilters();
```
Comment (jorgefilipecosta) [inline packages/edit-site/src/posts.js:41]: This code seems to be a duplication of packages/edit-site/src/index.js, could we create some function used in both places to avoid the duplication?
I fear in the future someone may easily change code in a place forgetting about the other place.

Comment (youknowriad) [reply packages/edit-site/src/posts.js:41]: I think duplication is fine sometimes, as it reminds us that this code is unnecessary boilerplate, it's not clear why we need to call it here in the first place.

Comment (jorgefilipecosta) [reply packages/edit-site/src/posts.js:41]: I guess if we are duplicating at least we should include comments in both files saying they are duplication of each other. Duplication ends up leading to divergence and we may end up with the double of the code to maintain.
I think code like block registration etc may be needed because the preview of the post depends on the blocks being registered and the editor being properly initialized.

**Review by jorgefilipecosta — COMMENTED** [review]
Things are working mostly well there are only some minor todo items pending. I think we should add a posts item in the site editor. It is very easy to run into a different site editor menu that then does not allow to go back to posts.

For example,open a post, edit the template of that post and press the back button. It is strange that a posts menu item is not there. Also, the items there like pages styles etc don't work at all.

That makes me wonder if instead of following this path, where we have a different initialization initializePostsDashboard, We could not just add a Posts tab in the site editor, and use the normal initializeEditor( id, settings ) as initialization given that we have a postType=post URL param the posts section would open right away. If we don't want to expose the posts section generally yet we could have a flag that just makes the section visible if the URL parameter postType=post was set when the component mounted.

**Comment by youknowriad** [comment]
@jorgefilipecosta I agree it's easier if this was developed as part of the site editor but that's not the UX experience we're targetting at this point. The idea is to replace the "posts" menu in wp-admin top level as otherwise the site editor will lose its "design" meaning. 

The fact that it's in a separate experimental page is temporary until we reach parity with the posts section.

File: packages/dataviews/src/dataviews.tsx (hunk 15 above)
Comment (ntsekouras) [reply packages/dataviews/src/dataviews.tsx:54]: It's unrelated but intentional. We had this duplicate function in the same package with no reason. I could remove from this PR, but I think it's harmless and super straightforward.

File: lib/experimental/posts/load.php (hunk 14 above)
Comment (ntsekouras) [reply lib/experimental/posts/load.php:71]: We need to have the param in the URL for the editor to resolve the current entities etc..

**Comment by jorgefilipecosta** [comment]
> @jorgefilipecosta I agree it's easier if this was developed as part of the site editor but that's not the UX experience we're targetting at this point. The idea is to replace the "posts" menu in wp-admin top level as otherwise the site editor will lose its "design" meaning.
> 
> The fact that it's in a separate experimental page is temporary until we reach parity with the posts section.

Hi @youknowriad, I understand that but I'm not convinced we need a different initialization. We could for example use exactly the same one, with a setting/flag to hide the other sections besides posts. Posts would be a normal section. So if the flag was not there the sidebar would render normal sections patterns, templates etc but not posts, with the flag there it would only include the posts section for now, given that we have a ?postType=posts we would automatically load with the post section open by default.


I guess we also need to decide what happens when the user goes to edit a template or pattern and presses the back button. Should we show the normal site editor sections if yes we need to make them work well, if not I guess we should add some logic on the back to on that case go back to the posts sidebar.

File: packages/dataviews/src/dataviews.tsx (hunk 15 above)
Comment (jorgefilipecosta) [reply packages/dataviews/src/dataviews.tsx:54]: Got it feel free to leave it.

File: lib/experimental/posts/load.php (hunk 14 above)
Comment (jorgefilipecosta) [reply lib/experimental/posts/load.php:71]: Maybe we could have a small js code on loading adding postType=post with history.push({postType:'post'}) if a postType is not set.

**Comment by youknowriad** [comment]
> Hi @youknowriad, I understand that but I'm not convinced we need a different initialization. We could for example use exactly the same one, with a setting/flag to hide the other sections besides posts. Posts would be a normal section. So if the flag was not there the sidebar would render normal sections patterns, templates etc but not posts, with the flag there it would only include the posts section for now, given that we have a ?postType=posts we would automatically load with the post section open by default.

I again agree with you that it's easiest to do this. But for me, doing a separately initialization is better because it actually force us to think about the initialization code base and remove it / simplify it basically. IMO, it shouldn't be necessary most of the time. It's either in the wrong place, not using the right API, not collocated in the right component... and using a separate initialization function force us to look at this gaps that we have and fix them.

> I guess we also need to decide what happens when the user goes to edit a template or pattern and presses the back button. Should we show the normal site editor sections if yes we need to make them work well, if not I guess we should add some logic on the back to on that case go back to the posts sidebar.

That's a good point and one we need to solve regardless of how we initialize this. I think there's probably a short term and a long term reply to this:

 - short term version is keep the behavior we currently have in the "post editor". "W" button always returns to the posts list.
 - long term version: do not touch the "W" button but navigating to a template or template part actually navigates to the site editor's template and template part. This can only work if everything is in a single SPA.

File: packages/edit-site/src/components/block-editor/use-site-editor-settings.js [hunk 18]
```diff
@@ -136,7 +136,10 @@ export function useSpecificEditorSettings() {
 				templateSlug: _record.slug,
 				canvasMode: getCanvasMode(),
 				settings: getSettings(),
-				postWithTemplate: _context?.postId,
+				postWithTemplate:
```
Comment (youknowriad) [inline packages/edit-site/src/components/block-editor/use-site-editor-settings.js]: "posts" actually have template, so I would change the value of this flag personally.

 - So either we should rename to something about `shouldUseTemplateAsDefaultRenderingMode` if it's only used for that config.
 - Move the "post type" check to later (when setting the `defaultRenderingMode` variable.

File: packages/edit-site/src/components/editor/index.js [hunk 19]
```diff
@@ -217,9 +218,20 @@ export default function EditSiteEditor() {
...
 										className="edit-site-layout__view-mode-toggle"
-										onClick={ () =>
-											setCanvasMode( 'view' )
-										}
+										onClick={ () => {
+											setCanvasMode( 'view' );
+											// TODO: this is a temporary solution to navigate to the posts list if we are
+											// come here throught `posts list` and are in focus mode editing a template or template part.
+											if (
+												isPostsList &&
```
Comment (youknowriad) [inline packages/edit-site/src/components/editor/index.js:226]: I think that's fine but we should be careful about not using this variable in a lot of places.

File: packages/edit-site/src/components/block-editor/use-site-editor-settings.js (hunk 18 above)
Comment (ntsekouras) [reply packages/edit-site/src/components/block-editor/use-site-editor-settings.js]: Makes sense to rename and was only used for the rendering mode selection, so I renamed to `shouldUseTemplateAsDefaultRenderingMode.

**Comment by youknowriad** [comment]
One thing that is unclear to me is what to do in the "home page" of the posts section. Right now it's showing the site's home page like the site editor but I feel that's not idea. 

That said I think we can merge this PR now. There's a lot of design questions that we need to follow-up with.

**Review by youknowriad — APPROVED** [review]

**Comment by ntsekouras** [comment]
>One thing that is unclear to me is what to do in the "home page" of the posts section. Right now it's showing the site's home page like the site editor but I feel that's not idea.

For me the ideal thing is to show a placeholder like many other similar apps/views do which could be something along `Please select something to preview` and/or `nothing is selected`.

**Comment by jorgefilipecosta** [comment]
> One thing that is unclear to me is what to do in the "home page" of the posts section. Right now it's showing the site's home page like the site editor but I feel that's not idea.
> 
> That said I think we can merge this PR now. There's a lot of design questions that we need to follow-up with.

Hi @youknowriad, I think if there is a blog front page it should show it.

**Comment by ellatrix** [comment]
Curious what happened here that affected the navigate metric so positively. Also the metric is more unstable now.

**Comment by ntsekouras** [comment]
>Curious what happened here that affected the navigate metric so positively. Also the metric is more unstable now.

This PR loads the post list under a new admin page and shouldn't affect any existing tests. 🤔

**Comment by ellatrix** [comment]
I'm talking about the pages list, that one was touched in this PR, no?
